### ISSUE TITLE
Hybrid + BitNet (I2_S) base path broken: 'Fused decode does not support I2_S'

### ISSUE THREAD

**Issue by jamesburton** [jamesburton/dotLLM#83]
## Bug
Running a BitNet b1.58 (I2_S) GGUF in **hybrid** mode (`--device gpu --gpu-layers N` with partial offload) fails:
- Decode throws `Fused decode does not support I2_S. Use standard Gemm path.`
- Prefill produces zero-norm logits (argmax 0 vs CPU's correct 5).

## Root cause
`HybridTransformerModel`'s CPU-phase decode fast path gates as `if (seqLen == 1 && _threadPool != null)` and calls `FusedQkvDecode`, which does not support I2_S. The full-CPU `TransformerModel` guards this with `&& lw.QQuantType != QuantizationType.I2_S` (routing ternary weights through the unfused I2_S GEMV path); `HybridTransformerModel` lacks that guard. The GPU phase may also need I2_S coverage in its `ProjectGpu` decode path.

## Repro
`dotnet run --project src/DotLLM.Cli -- run <bitnet-i2_s.gguf> --prompt "The capital of France is" --device gpu --gpu-layers 5`

## Scope / impact
Pre-existing — predates the Hybrid LoRA work (#82). BitNet works correctly on full `--device cpu` and full `--device gpu`; only the hybrid split is affected. Discovered while building #82's parity test (which was therefore re-pointed to a non-BitNet model).

## Fix sketch
Mirror `TransformerModel`'s I2_S gating in `HybridTransformerModel`'s CPU decode condition (and verify the GPU `ProjectGpu` decode path dispatches the I2_S GEMV). Add a hybrid+BitNet smoke/parity test.

**Comment by jamesburton** [comment]
## Root-cause analysis (systematic debugging)

Two (likely three) distinct root causes, not one:

1. **CPU decode throws** — `HybridTransformerModel` CPU decode conditions (the `if (seqLen==1 && _threadPool!=null && !adapterActive)` at the attention + FFN blocks) lack the full-CPU model's `&& lw.QQuantType != QuantizationType.I2_S` guard, so I2_S weights hit `MatMul.FusedDecodeGemv3/2` which throws `"Fused decode does not support {qt}"` (`MatMulFusedDecode.cs:431`).

2. **GPU-phase prefill zero/NaN logits** — `CudaTransformerModel` constructs its state with `useFp32Residual = (Architecture == BitNet)` (`CudaTransformerModel.cs:426`) and routes residual junctions through FP32 (`LaunchCopyF16ToF32`, `LaunchFusedAddRmsNormF32Res`, `LaunchAddF32F16`, final `LaunchRmsNormF32InF16W`) because BitNet's residual magnitude exceeds FP16's ~65504 ceiling. The hybrid GPU loop uses FP16 `_gpuState.Residual` with `LaunchFusedAddRmsNorm`/`LaunchAdd` and constructs `_gpuState` without `useFp32Residual` → FP16 overflow → degenerate output.

3. **GPU→CPU boundary transfer is FP16** — the hybrid boundary copies `_gpuState.HiddenState` (FP16) via `_fp16TransferBuffer`. For BitNet the un-normalized residual at the layer boundary may exceed FP16 range, so a BitNet-specific **FP32** boundary transfer (D2H of `ResidualF32` → CPU FP32 hidden) is likely required.

## Fix
Mirror `CudaTransformerModel`'s BitNet FP32-residual path into the hybrid GPU loop (+ FP32 boundary transfer for BitNet) and add the I2_S decode guard to the hybrid CPU loop. Gate with a hybrid+BitNet parity test vs the full-CPU path. Being fixed in branch `fix/hybrid-bitnet-i2s`.

**Comment by jamesburton** [comment]
Fixed — merged to dev via PR #85 (FP32 residual + FP32 boundary + I2_S decode guard + I2_S prefill dispatch; hybrid+BitNet parity vs full-CPU cosine 0.999995). BitNet now works across CPU/GPU/hybrid. Closed manually since the PR targeted dev.
